lib/models/message_container.dart

```diff
@@ -1,240 +0,0 @@
-          : copyWith(
-              selfIdMessage: message as SelfIDMessage,
-              lastMessageRssi: rssi,
-              lastUpdate:
-                  DateTime.fromMillisecondsSinceEpoch(receivedTimestamp),
-              source: source,
-            ),
-      OperatorIDMessage => operatorIdMessage != null &&
-              operatorIdMessage!.containsEqualData(message as OperatorIDMessage)
-          ? null
-          : copyWith(
-              operatorIdMessage: message as OperatorIDMessage,
-              lastMessageRssi: rssi,
-              lastUpdate:
-                  DateTime.fromMillisecondsSinceEpoch(receivedTimestamp),
-              source: source,
-            ),
-      AuthMessage => authenticationMessage != null &&
-              authenticationMessage!.containsEqualData(message as AuthMessage)
-          ? null
-          : copyWith(
-              authenticationMessage: message as AuthMessage,
-              lastMessageRssi: rssi,
-              lastUpdate:
-                  DateTime.fromMillisecondsSinceEpoch(receivedTimestamp),
-              source: source,
-            ),
-      SystemMessage => systemDataMessage != null &&
-              systemDataMessage!.containsEqualData(message as SystemMessage)
-          ? null
-          : copyWith(
-              systemDataMessage: message as SystemMessage,
-              lastMessageRssi: rssi,
-              lastUpdate:
-                  DateTime.fromMillisecondsSinceEpoch(receivedTimestamp),
-              source: source,
-            ),
-      _ => null
-    };
-  }
-
-  pigeon.MessageSource get packSource => source;
-
-  bool get operatorIDSet =>
-      operatorIdMessage != null &&
-      operatorIdMessage!.operatorID != OPERATOR_ID_NOT_SET;
-
-  bool get operatorIDValid {
-    final validCharacters = RegExp(r'^[a-zA-Z0-9]+$');
-    return operatorIdMessage != null &&
-        operatorIdMessage!.operatorID.length == 16 &&
-        validCharacters.hasMatch(operatorIdMessage!.operatorID);
-  }
-
-  bool get systemDataValid =>
-      systemDataMessage != null &&
-      systemDataMessage?.operatorLocation != null &&
-      systemDataMessage!.operatorLocation!.latitude != INV_LAT &&
-      systemDataMessage?.operatorLocation!.longitude != INV_LON &&
-      systemDataMessage!.operatorLocation!.latitude <= MAX_LAT &&
-      systemDataMessage!.operatorLocation!.latitude >= MIN_LAT &&
-      systemDataMessage!.operatorLocation!.longitude <= MAX_LON &&
-      systemDataMessage!.operatorLocation!.longitude >= MIN_LON;
-
-  bool get locationValid =>
-      locationMessage != null &&
-      locationMessage?.location != null &&
-      locationMessage!.location!.latitude != INV_LAT &&
-      locationMessage!.location!.longitude != INV_LON &&
-      locationMessage!.location!.latitude <= MAX_LAT &&
-      locationMessage!.location!.longitude <= MAX_LON &&
-      locationMessage!.location!.latitude >= MIN_LAT &&
-      locationMessage!.location!.longitude >= MIN_LON;
```

Comment:
We lost some validation & filtering this way, should we add it back somehow? For example as a function that could filter the stream and can be used by the target app?

Imagine something like:

```dart
  flutterOpendroneId.receivedMessages.map(filterInvalidMessages).listen( ...
```

And it would be completely optional this way, and also more transparent, from the look in the target app, you would understand that it's doing some filtering.

It also makes me think if this isn't a reponsibility of dart_opendroneid to provide validation methods/functions. WDYT @albertmoravec

Reply:
We could add those filters as a separate feature of dart_opendroneid. These checks are not in the spec (F3411), so I would not make them implicit, but as an optional feature it should be okay.